### PR TITLE
Format everything with Ormolu

### PR DESCRIPTION
**Pull request by nc6**
Since we were discussing formatting changes... here's a script that formats using ormolu, and an example of what this looks like.

I don't necessarily agree with all of the formatting decisions, but I do like:
- The fact that this defines a single format
- The handling of multi-line things. In particular, I just have to decide "one or many", and the formatter sorts out the rest. I feel like if I'm going to have a formatter, it's useless if I also have to spend a bunch of time thinking about the formatting as well...

Obviously we should not merge this actual PR unless the repo is quiescent, since it will conflict a lot. So this is mostly for discussion

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs [hunk 1]
```diff
@@ -174,29 +184,29 @@ data StrictMaybe a
 instance NoUnexpectedThunks a => NoUnexpectedThunks (StrictMaybe a)
 
 instance Functor StrictMaybe where
-  fmap _ SNothing  = SNothing
+  fmap _ SNothing = SNothing
```
Comment (JaredCorduan) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs:187]: I do like sometimes aligning on the `=`, but this seems like a small price to pay for consistency.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs [hunk 2]
```diff
@@ -240,68 +250,74 @@ text64FromCBOR = do
...
-                      -- In this degenerate case, where ln (1-f) is not defined,
-                      -- we set the unActiveSlotLog to zero.
-                      then 0
-                      else floor (fpPrecision * (
-                        ln' $ (1 :: FixedPoint) - (fromRational $ intervalValue v))) }
+  ActiveSlotCoeff
+    { unActiveSlotVal = v,
+      unActiveSlotLog =
+        if (intervalValue v) == 1
+          then-- If the active slot coefficient is equal to one,
```
Comment (nfrisby) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs:309]: Random drive-by commentary: `then--` is weird

Comment (nc6) [reply shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs:309]: Yep, I mentioned this to Mark, he said it's a known bug that will be fixed as part of the current project :-)

**Comment by uroboros** [comment]
I also like this - although there are a few things I'll miss for a while until the new habit sets in, e.g. leading commas, and left aligned (::), (->), (=>) - but that's not an objection.

**Comment by nc6** [comment]
Closing since this PR was for showing the format. We'll open a separate PR to apply the formatting at a time when the codebase is stable.
